### ISSUE TITLE
MSIX packaging for arbitrary .NET project types

### ISSUE THREAD

**Issue by daniel-lerch** [microsoft/WindowsAppSDK#4932]
### Motivation

The .NET SDK supports streamlined MSIX publishing for [WinUI 3](https://learn.microsoft.com/de-de/windows/apps/windows-app-sdk/single-project-msix?tabs=csharp) and [MAUI](https://learn.microsoft.com/en-us/dotnet/maui/windows/deployment/publish-cli?view=net-maui-9.0) projects. For other project types one can use an [application packaging project](https://learn.microsoft.com/de-de/windows/msix/desktop/desktop-to-uwp-packaging-dot-net). Packaging projects are cumbersome to use, more complicated to use than `dotnet` in CI pipeline and cannot pass MSBuild properties to child projects easily.

### Desired solution

I would like to have MSIX packaging tools for arbitrary .NET project types including Avalonia, WPF, WinForms, Console. It should be compatible with other features like single file executable, trimming and Native AOT.

### Additional context

I originally posted this issue at dotnet/sdk#45045. However, as @Sergio0694 pointed out, MSIX packaging functionality is already implemented in `Microsoft.Windows.SDK.BuildTools.MSIX` and could be extended to support other frameworks than WinUI 3 or MAUI. Unfortunately, I could not find the source code for that NuGet package and hope that I am not totally wrong by opening this issue here.

**Comment by Sergio0694** [comment]
> "could be extended to support other frameworks than WinUI 3 or MAUI"

Note, UWP (on .NET 9) also uses that package, so it's already the case that it supports several different frameworks. The MSIX tooling itself doesn't really care about the UI framework in the first place. Have you tried using it with an Avalonia/WPF/WinForms app already, as a test? Did you get any errors in particular? Knowing that could be useful as a starting point 🙂

cc. @Scottj1s

**Comment by daniel-lerch** [comment]
Single project MSIX packaging works „out of the box“ for WinForms 🤯

I created a _Blank App, Packaged (WinUI 3 in Desktop)_ and copied over the `Assets` folder, `Properties` folder including `launchSettings.json` and `PublishProfiles` and almost the entire project file to a Windows Forms project and it works in Visual Studio.

When I tested this before opening this issue I used only the online documentation for MAUI MSIX packaging and did not use samples from project templates which did not work. Looking forward I will test more frameworks and configurations:

- [x] WinForms
- [ ] WPF
- [ ] Avalonia
- [ ] CI/CD Integration

If everything works, I will close this issue and open issues at the respective docs repositories to inform developers about this amazing feature.

**Comment by Sergio0694** [comment]
This aligns with what I was expecting, and it was one of the reasons why we decoupled the MSIX tooling in its own package 🙂
In theory you should be able to just add a package reference to it, set `EnableMsixTooling`, and that's it. Don't even need WinAppSDK.

**Comment by DarranRowe** [comment]
@Sergio0694 

Just to clarify, is this an official statement that this package is supported in arbitrary projects? If it is, where can we find the documentation, or if it doesn't exist yet, when can we expect the documentations?

**Comment by Sergio0694** [comment]
It's not an official statement, no. Right now, the package is only officially supported for UWP, and it will be officially supported for WinUI 3 when WinAppSDK switches to it as well in a future release. I will see if we can add documentation to make this overall support official though. I'm just saying in the meantime, people can try it out on other frameworks (eg. WinForms, WPF), and it should work fine, in theory 🙂

**Comment by daniel-lerch** [comment]
> You should be able to just add a package reference to it, set `EnableMsixTooling`, and that's it. Don't even need WinAppSDK.

That's right. I can build and deploy arbitrary projects as MSIX using Visual Studio's launch configuration.

However, I faced two problems:
1. Visual Studio's package and publish context menu works only when using `Microsoft.WindowsAppSdk` instead of `Microsoft.Windows.SDK.BuildTools.MSIX`. Setting `HasPackageAndPublishMenu` and `<ProjectCapability Include="Msix" />` have no effect on that.
2. When I build for x86 the first time, I get an error (MSB4062) which disappears once I built for x64 which does not cause this error.<br>
The "Microsoft.Build.Msix.Recipe.WinAppSdkMatchItemsByFileNameOnMetadataValue" task could not be loaded from the assembly C:\Users\***\.nuget\packages\microsoft.windows.sdk.buildtools.msix\1.1.20241024.1\build\..\tools\net472\Microsoft.Build.Msix.dll.  Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask.<br>
I will try to reproduce this error within the next days.
